state/migrations/20230822180807_bogus_snapshot_cleanup.go

```diff
@@ -0,0 +1,93 @@
+package migrations
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"github.com/lib/pq"
+	"github.com/pressly/goose/v3"
+	"github.com/rs/zerolog"
+	"os"
+)
+
+var logger = zerolog.New(os.Stdout).With().Timestamp().Logger().Output(zerolog.ConsoleWriter{
+	Out:        os.Stderr,
+	TimeFormat: "15:04:05",
+})
+
+func init() {
+	goose.AddMigrationContext(upBogusSnapshotCleanup, downBogusSnapshotCleanup)
+}
+
+func upBogusSnapshotCleanup(ctx context.Context, tx *sql.Tx) error {
+	// Run a one-off script to delete rooms with bogus snapshots.
+	// "Bogus" means "lacking a create event".
+	bogusRooms, err := getBogusRooms(ctx, tx)
+	if err != nil {
+		return err
+	}
+	if len(bogusRooms) == 0 {
+		return nil
+	}
+	logger.Info().Strs("room_ids", bogusRooms).
+		Msgf("Found %d bogus rooms to cleanup", len(bogusRooms))
+
+	tables := []string{"syncv3_snapshots", "syncv3_events", "syncv3_rooms"}
+	for _, table := range tables {
+		err = deleteFromTable(ctx, tx, table, bogusRooms)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func deleteFromTable(ctx context.Context, tx *sql.Tx, table string, roomIDs []string) error {
+	result, err := tx.ExecContext(
+		ctx,
+		`DELETE FROM `+table+` WHERE room_id = ANY($1)`,
+		pq.StringArray(roomIDs))
+	if err != nil {
+		return fmt.Errorf("failed to delete from %s: %w", table, err)
+	}
+	ra, err := result.RowsAffected()
+	if err != nil {
+		logger.Warn().Err(err).Msgf("Couldn't get number of rows deleted from %s", table)
+	} else {
+		logger.Info().Msgf("Deleted %d snapshots from %s", ra, table)
```

Comment:
```suggestion
		logger.Info().Msgf("Deleted %d rows from %s", ra, table)
```
Technically they aren't always snapshots